qcodes/instrument_drivers/rigol/DG1062.py

```diff
@@ -290,7 +304,32 @@ def _set_waveform_params(self, **params_dict: Dict) ->None:
         string += ",".join(
             ["{:7e}".format(params_dict[param]) for param in param_names])
         self.parent.write_raw(string)
+        
+    def _get_duty_cycle(self):
+        """
+        Simple function to extract the duty cycle after checking waveform
+        """
+        self.waveform()
+        wf = self.waveform.get_latest()
+
+        if wf == 'PULS' or wf == 'SQU':
+            duty_cycle = self.parent.ask_raw(f":SOUR{self.channel}:FUNC:{wf}:DCYC?")
```

Comment:
It is quite weird that throughout this driver `self.parent.ask_raw` and `self.parent.write_raw` are used instead of just `self.ask` and `self.write` which do the same, but are much cleaner to use from the code/design point of view.

For this driver and this PR, ok, let's keep it consistent. But in general, it's expected that `self.write` and `self.ask` of the `InstrumentChannel` (which is the class of `self` here) class are used, unless there is some really important reason not to do it.

Reply:
to be honest, i tried to stay consistent within the code. maybe something for the next merge?

Reply:
sure.